### PR TITLE
Fix UserGearController

### PR DESCRIPTION
**Pull request by EllieLilithFox**
All methods on the controller now work. Changed UserId to string type.

### CODE REVIEW

File: ClimbingGearBackend/Controllers/UserGearController.cs [hunk 1]
```diff
@@ -20,10 +20,9 @@ public UserGearController(ClimbingGearContext context)
 
     // GET: api/UserGear
     [HttpGet]
-    public async Task<ActionResult<IEnumerable<UserGearDTO>>> GetUserGear(long userId)
+    public async Task<ActionResult<IEnumerable<UserGearDTO>>> GetUserGear()
```
Comment (NatalieWolfe) [inline ClimbingGearBackend/Controllers/UserGearController.cs:24]: What is the reason for this endpoint? Allowing anyone to see everyone's gear doesn't make sense.

File: ClimbingGearBackend/Controllers/UserGearController.cs [hunk 2]
```diff
@@ -38,11 +37,11 @@ public async Task<ActionResult<IEnumerable<UserGearDTO>>> GetUserGear(long userI
 
     // GET: api/UserGear/5
     [HttpGet("{id}")]
-    public async Task<ActionResult<UserGearDTO>> GetSingleUserGear(long userGearId)
+    public async Task<ActionResult<UserGearDTO>> GetUserGear(long id)
```
Comment (NatalieWolfe) [inline ClimbingGearBackend/Controllers/UserGearController.cs:41]: Should there be some validation that the authenticated user has permission to view this particular item? Similarly for the mutation methods below.

File: ClimbingGearBackend/Migrations/ClimbingGearContextModelSnapshot.cs [hunk 3]
```diff
@@ -70,8 +70,8 @@ protected override void BuildModel(ModelBuilder modelBuilder)
                     b.Property<int>("Quantity")
                         .HasColumnType("integer");
 
-                    b.Property<long>("UserId")
-                        .HasColumnType("bigint");
+                    b.Property<string>("UserId")
+                        .HasColumnType("text");
```
Comment (NatalieWolfe) [inline ClimbingGearBackend/Migrations/ClimbingGearContextModelSnapshot.cs]: Why is this text? Isn't the UserId a number?

**Review by NatalieWolfe — CHANGES_REQUESTED** [review]

File: ClimbingGearBackend/Controllers/UserGearController.cs (hunk 1 above)
Comment (EllieLilithFox) [reply ClimbingGearBackend/Controllers/UserGearController.cs:24]: I just wanted it in a working state so I know it works. It's a simple matter(I mean it has to be simple to someone) to lock the whole controller and add a query that looks for the current user and returns their gear.

File: ClimbingGearBackend/Controllers/UserGearController.cs (hunk 2 above)
Comment (EllieLilithFox) [reply ClimbingGearBackend/Controllers/UserGearController.cs:41]: Yes, I am going to add that. The way that piece of code works is that the id is grabbed from the route so `api/usergear/5` would pass `5` as a parameter to the method. If the parameter names for don't match it won't get passed to the method.

File: ClimbingGearBackend/Migrations/ClimbingGearContextModelSnapshot.cs (hunk 3 above)
Comment (EllieLilithFox) [reply ClimbingGearBackend/Migrations/ClimbingGearContextModelSnapshot.cs]: No, Identity uses UUIDs for their user IDs. So they are all strings.

File: ClimbingGearBackend/Controllers/UserGearController.cs (hunk 2 above)
Comment (NatalieWolfe) [reply ClimbingGearBackend/Controllers/UserGearController.cs:41]: Often, things you intend to add in the future which are necessary for completeness like this should be noted with a `TODO` comment. This way reviewers know you have thought about it and intend to return to it.

**Pull request merged**